src/core/ready.js

```diff
@@ -1,17 +1,16 @@
 define( [
 	"../core",
-	"../var/document",
-	"../deferred"
+	"../var/document"
 ], function( jQuery, document ) {
 
-// The deferred used on DOM ready
-var readyList;
+var readyCallbacks = [],
+	readyFiring = false,
+	whenReady = function( fn ) {
+		readyCallbacks.push( fn );
```

Comment:
Agreed, given that it's inexpensive, I'd want to do whichever is smaller. Right now, that would be what we currently have.